python-project/mortar/image/threshold.py

```diff
@@ -16,25 +15,17 @@
 class Threshold(Filter):
     """
     Perform thresholding on an image.
-
-    TODO this class doesn't yet support the necessary thresholding options.
     """
     name = 'Threshold'
 
-    def run(self, input: PILImage) -> Optional[PILImage]:
+    def run(self,
```

Comment:
The pipeline composes any number of Filter objects and doesn't know their derived types. It calls the `run` method with the signature in the base `Filter` class, so this won't work there; the arguments all get their default values. See the `pipeline.py`.

Can you move the arguments to the constructor, store their values in some class attributes, and then use those values from the `run` method? See the `Crop` filter for an existing example of parametrization.